x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/log_categorization_for_discover.tsx

```diff
@@ -0,0 +1,472 @@
+    const { getIndexPattern, timeFieldName: timeField } = dataView;
+    const index = getIndexPattern();
+
+    if (
+      loading === true ||
+      selectedField === null ||
+      timeField === undefined ||
+      earliest === undefined ||
+      latest === undefined ||
+      minimumTimeRangeOption === undefined ||
+      mounted.current !== true
+    ) {
+      return;
+    }
+
+    cancelRequest();
+
+    setLoading(true);
+    setData(null);
+    setFieldValidationResult(null);
+
+    const additionalFilter: CategorizationAdditionalFilter = {
+      from: earliest,
+      to: latest,
+    };
+
+    const runtimeMappings = dataView.getRuntimeMappings();
+
+    try {
+      const timeRange = await getMinimumTimeRange(
+        index,
+        timeField,
+        additionalFilter,
+        minimumTimeRangeOption,
+        searchQuery,
+        runtimeMappings
+      );
+
+      if (mounted.current !== true) {
+        return;
+      }
+
+      const [validationResult, categorizationResult] = await Promise.all([
+        runValidateFieldRequest(
+          index,
+          selectedField.name,
+          timeField,
+          timeRange,
+          searchQuery,
+          runtimeMappings,
+          {
+            [AIOPS_TELEMETRY_ID.AIOPS_ANALYSIS_RUN_ORIGIN]: embeddingOrigin,
+          }
+        ),
+        runCategorizeRequest(
+          index,
+          selectedField.name,
+          timeField,
+          { to: timeRange.to, from: timeRange.from },
+          searchQuery,
+          runtimeMappings,
+          intervalMs,
+          timeRange.useSubAgg ? additionalFilter : undefined
+        ),
+      ]);
+
+      if (mounted.current !== true) {
+        return;
+      }
+
+      setFieldValidationResult(validationResult);
+      const { categories, hasExamples } = categorizationResult;
+
+      if (timeRange.useSubAgg) {
+        const categoriesInBucket = categorizationResult.categories
+          .map((category) => ({
+            ...category,
+            count: category.subFieldCount ?? category.subTimeRangeCount!,
+            examples: category.subFieldExamples!,
+            sparkline: category.subFieldSparkline,
+          }))
+          .filter((category) => category.count > 0)
+          .sort((a, b) => b.count - a.count);
+        setData({
+          categories: categoriesInBucket,
+          displayExamples: hasExamples,
+          totalCategories: categories.length,
+        });
+      } else {
+        setData({
+          categories,
+          displayExamples: hasExamples,
+          totalCategories: categories.length,
+        });
+      }
+    } catch (error) {
+      if (error.name === 'AbortError') {
+        // ignore error
+      } else {
+        toasts.addError(error, {
+          title: i18n.translate('xpack.aiops.logCategorization.errorLoadingCategories', {
+            defaultMessage: 'Error loading categories',
+          }),
+        });
+      }
+    }
```

Comment:
nit: 
```suggestion
      if (error.name !== 'AbortError') {
        toasts.addError(error, {
          title: i18n.translate('xpack.aiops.logCategorization.errorLoadingCategories', {
            defaultMessage: 'Error loading categories',
          }),
        });
      }
    }
```

Reply:
Updated in https://github.com/elastic/kibana/pull/186539/commits/a3008dc3d0f23cb78be2899bcb59236eec21d846